### ISSUE TITLE
500 Internal Error 

### ISSUE THREAD

**Issue by s3m1s0n1c** [irvinlim/apple-health-ingester#24]
Hey Guys,

Great Project.. Thanks for your time...

Im getting this error when exporting using "Auto Export" app using Free Trail period..

error for InfluxDB: unmarshal error: []*healthautoexport.Datapoint: unmarshalerDecoder: gghealthautoexport.datapointCopy.Qty: Date: unmarshalerDecoder: parsing time "\"2024-09-21 7:57:00\xe2\x80\xafam +1000\"" as "\"2006-01-02 15:04:05 -0700\"": cannot parse "\xe2\x80\xafam +1000\"" as " ", error found in #10 byte of ...|am +1000",
        |..., bigger context ...|          "date" : "2024-09-21 7:57:00 am +1000",
            "qty" : 0.049035839369578754,
      |..., error found in #10 byte of ...|         },
        |..., bigger context ...|39369578754,
            "source" : ""
          },
          {
            "date" : "2024-09-21 7:5|....
Try again or contact support if this issue continues.

I cant for the life of me figure it out..

Maybe the data has changed from AutoExport?

Thanks
Sonic

**Comment by s3m1s0n1c** [comment]
I think the payload data has changed which is causing this error..

anyone?

**Comment by irvinlim** [comment]
Sorry for lack of replies. Let me check this when I have time over the weekend. 

The timestamp is coming back weird, there's some additional non-ASCII characters in the returned timestamp.

Can I check what's the _Region_ setting on your phone? (Under _General_ > _Language & Region_)

**Comment by s3m1s0n1c** [comment]
Thanks for the quick reply,

Im From Australia..

Here is a Screeny of my settings

![image](https://github.com/user-attachments/assets/f54a09eb-a974-40c2-b0f9-74ff6da01dae)

Thanks Sonic

**Comment by s3m1s0n1c** [comment]
Oh I think i figured it out..

It needs Phone to be on 24 hr time :)

```2024/10/04 09:02:26 Total metrics: 1 (1 populated) Total samples 132
2024/10/04 09:02:26 Starting upload to metric store "influxdb".
```

**Comment by s3m1s0n1c** [comment]
Yep confirmed that Phone time **NEEDS TO BE IN 24 HR TIME** for this ingester to work

Thanks
Sonic

**Comment by s3m1s0n1c** [comment]
Yep confirmed that Phone time NEEDS TO BE IN 24 HR TIME for this ingester to work

Thanks
Sonic

**Comment by irvinlim** [comment]
Thanks for the analysis! I'll create a new issue to support 12-hour time as well.

Actually the big issue here is that the timestamp being locale specific in the response makes it near impossible to support all locales...
